apps/astarte_appengine_api/test/astarte_appengine_api/device/device_reading_v2_test.exs

```diff
@@ -0,0 +1,299 @@
+# limitations under the License.
+
+defmodule Astarte.AppEngine.API.Device.DeviceReadingV2Test do
+  use Astarte.Cases.Data, async: false
+
+  use Astarte.Cases.Device
+  use ExUnitProperties
+
+  alias Astarte.AppEngine.API.Device
+  alias Astarte.AppEngine.API.Device.InterfaceValues
+  alias Astarte.DataAccess.Realms.Realm
+  alias Astarte.DataAccess.Repo
+
+  import Astarte.Helpers.Device
+  import Astarte.InterfaceUpdateGenerators
+
+  describe "get_interface_value" do
+    property "returns the value (from root) ", context do
+      %{
+        realm_name: realm_name,
+        astarte_instance_id: astarte_instance_id,
+        interfaces: interfaces,
+        device: device
+      } = context
+
+      clean_devices_saved_data(astarte_instance_id, realm_name)
+
+      on_exit(fn ->
+        clean_devices_saved_data(astarte_instance_id, realm_name)
+      end)
+
+      valid_interfaces_for_update = interfaces |> Enum.filter(&(&1.ownership == :server))
+
+      check all(
+              interface_to_update <- member_of(valid_interfaces_for_update),
+              mapping_update <- valid_mapping_update_for(interface_to_update)
+            ) do
+        update_value = mapping_update.value
+        path_tokens = String.split(mapping_update.path, "/")
+        expected_token = [realm_name, device.encoded_id, interface_to_update.name | path_tokens]
+
+        expected_published_value =
+          expected_published_value!(mapping_update.value_type, update_value)
+
+        expected_qos = expected_qos_for!(mapping_update)
+
+        expected_read_value = expected_read_value!(mapping_update.value_type, update_value)
+        read_path = path_tokens |> Enum.drop(1)
+
+        publish_result_ok(interface_to_update, mapping_update, fn args ->
+          assert %{payload: payload, topic_tokens: topic_tokens, qos: qos} = args
+          assert topic_tokens == expected_token
+          assert qos == expected_qos
+          assert {:ok, %{"v" => ^expected_published_value}} = Cyanide.decode(payload)
+        end)
+
+        {:ok, _} =
+          Device.update_interface_values(
+            realm_name,
+            device.encoded_id,
+            interface_to_update.name,
+            mapping_update.path,
+            update_value,
+            []
+          )
+
+        {:ok, %InterfaceValues{data: result}} =
+          Device.get_interface_values!(
+            realm_name,
+            device.encoded_id,
+            interface_to_update.name,
+            %{}
+          )
+
+        result = get_in(result, read_path)
+        assert valid_result?(result, interface_to_update, expected_read_value)
+      end
+    end
+
+    property "returns nil", context do
+      %{
+        realm_name: realm_name,
+        astarte_instance_id: astarte_instance_id,
+        interfaces: interfaces,
+        device: device
+      } = context
+
+      clean_devices_saved_data(astarte_instance_id, realm_name)
```

Comment:
As @noaccOS said, tests should leave db in a clean state, so doing this here is discouraged